### PR TITLE
Update CDI/CDI-API vendoring to get 1.55.0

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
Currently KubeVirt uses CDI 1.50.0 and CDI-API 1.42.0 which is not consistent. This updates them both to 1.55.0

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Will need to update CDI version in kubevirtci image to also be 1.55.0

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/dep-update.sh [hunk 1]
```diff
@@ -23,26 +23,26 @@ done
     echo $_sync_only
     cd staging/src/kubevirt.io/api
     if [ "${_sync_only}" == "false" ]; then go get $@ ./...; fi
-    go mod tidy
+    go mod tidy -compat=1.17
```
Comment (akalenyu) [inline hack/dep-update.sh:26]: Last time around @mhenriks suggested that when we move to go 1.18, we can remove -compat=1.17 so maybe comment that

**Review by akalenyu — COMMENTED** [review]
Shouldn't we also update the vendored CDI API in the go.mod's of the various kubevirt APIs?

File: pkg/virt-operator/resource/apply/rbac_test.go [hunk 2]
```diff
@@ -285,6 +286,7 @@ var _ = Describe("RBAC test", func() {
 			required := newEmptyResource(resourceType)
 
 			assignRulesToRoles(newFakePolicyRules("policy1"), existing, required)
+			getRbacMetaObject(required).OwnerReferences = []metav1.OwnerReference{}
```
Comment (akalenyu) [inline pkg/virt-operator/resource/apply/rbac_test.go:289]: hmm I assumed the unit tests failures were CDI-related, interesting, what is happening here?

Comment (awels) [reply pkg/virt-operator/resource/apply/rbac_test.go:289]: I don't think they are CDI related, it is likely some k8s dependency that is now returning an empty list of ownerRefs vs a nil.

File: hack/dep-update.sh (hunk 1 above)
Comment (awels) [reply hack/dep-update.sh:27]: Added

**Comment by akalenyu** [comment]
Some bazel cache timeouts look like
https://coreos.slack.com/archives/C01EX3K1FGE/p1664308590913589

File: go.mod [hunk 3]
```diff
@@ -78,8 +78,8 @@ require (
 	k8s.io/utils v0.0.0-20211116205334-6203023598ed
 	kubevirt.io/api v0.0.0-00010101000000-000000000000
 	kubevirt.io/client-go v0.0.0-00010101000000-000000000000
-	kubevirt.io/containerized-data-importer v1.41.0
-	kubevirt.io/containerized-data-importer-api v1.50.0
+	kubevirt.io/containerized-data-importer v1.55.0
```
Comment (brybacki) [inline go.mod:81]: Where do we need the cdi (not the API)?

**Comment by brybacki** [comment]
CDI is already on 1.55 in kubevirtci 
https://github.com/kubevirt/kubevirtci/pull/878

File: go.mod (hunk 3 above)
Comment (awels) [reply go.mod:81]: Because kubevirt uses one or two functions from CDI (to determine some fields) to ensure CDI and kubevirt use the same calculation.

**Pull request merged**